src/printer.js

```diff
@@ -414,7 +414,11 @@ const embed = (path, print, textToDoc, opts) => {
     // the css parser appends an extra indented hardline, which we want outside of the `indent()`,
     // so we remove the last element of the array
     const [formatttedStyles, _] = textToDoc(node.content.styles, { ...opts, parser: parserLang });
-    return group([styleLang !== 'css' ? `<style lang="${styleLang}">` : '<style>', indent([hardline, formatttedStyles]), hardline, '</style>', hardline]);
+
+    const attributes = path.map((childPath) => childPath.call(print), 'attributes');
```

Comment:
i don't totally understand what's going on here - is it calling 'attribute printing' on every child of the node? 😅

Reply:
This is my understanding of how this works, please correct me if I'm wrong:

'path' and 'childPath' are AstPaths.
Calling  `AstPath.getValue()`  gives you what I call the current node. Calling `AstPath.call(print)` prints the current node.
Calling  `AstPath.map(callback, 'attributes')`  will loop the current node property 'attributes' and  "append" a new "current node" to the childPath.

```js
 const attributes = path.map((childPath) => {
      console.log(childPath.getValue()); // this will print a node of type 'Attribute' each time, like:
      // { start: 9, end: 15, type: 'Attribute', name: 'global', value: true }
      return childPath.call(print);  // this will print the current node
    }, 'attributes');
```

I based my code on [this lines of prettier-plugin-svelte](https://github.com/sveltejs/prettier-plugin-svelte/blob/master/src/embed.ts#L198).
This the code for [map](https://github.com/prettier/prettier/blob/main/src/common/ast-path.js#L105) from the prettier repo.

Reply:
Still don't quite understand, but that sounds reasonable 😅 and definitely a good idea to pull from the svelte prettier plugin, since we forked it!